### PR TITLE
Run the collector as a custom user 

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:** 
Have the collector run as a regular user instead of root user for security concern

**Testing:** 
Tested in Amazon Linux EC2

### CODE REVIEW

File: cmd/awscollector/main.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ import (
 // from opentelemetry-collector list
 func main() {
 	logger.SetupErrorLogger()
+
```
Comment (wyTrivail) [inline cmd/awscollector/main.go:37]: any new empty line?

File: pkg/userutils/userutil_darwin.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package userutils
+
+func ChangeUser() (user string, err error) {
+	return "", nil
```
Comment (wyTrivail) [inline pkg/userutils/userutil_darwin.go:21]: add an todo comment?

File: pkg/userutils/userutil_linux.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	"strconv"
+	"syscall"
+
+	"github.com/aws-observability/aws-otel-collector/pkg/logger"
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/opencontainers/runc/libcontainer/user"
+	"golang.org/x/sys/unix"
+)
+
+var defaultUser = "aoc"
```
Comment (wyTrivail) [inline pkg/userutils/userutil_linux.go:34]: maybe not using aoc?

File: pkg/userutils/userutil_linux.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	newUser, err := user.GetExecUserPath(runasuser, &currExecUser, "/etc/passwd", "/etc/group")
+	if err != nil {
+		log.Printf("E! Failed to get newUser: %v", err)
+		return nil, err
+	}
+	return newUser, nil
+}
+
+// ChangeUser allow customers to run the collector in selected user
+// by default it ran as 'aoc' user but can be set by environment variable
```
Comment (wyTrivail) [inline pkg/userutils/userutil_linux.go:78]: if you change aoc to some other user name, change the comment as well

File: tools/packaging/linux/aws-otel-collector.service [hunk 5]
```diff
@@ -23,8 +23,8 @@ After=network.target
 
 [Service]
 Type=simple
-User=aoc
-Group=aoc
+# User=aoc
```
Comment (wyTrivail) [inline tools/packaging/linux/aws-otel-collector.service:26]: change the user name?

**Review by wyTrivail — CHANGES_REQUESTED** [review]

**Review by wyTrivail — APPROVED** [review]

File: pkg/userutils/userutil_linux.go (hunk 3 above)
Comment (mxiamxia) [reply pkg/userutils/userutil_linux.go:35]: yes, will update it in next pr

File: pkg/logger/logger.go [hunk 6]
```diff
@@ -80,9 +75,9 @@ func SetupErrorLogger() {
 // getLogFilePath retuns the log file path depending on the OS.
 func getLogFilePath() string {
 	if runtime.GOOS == "windows" {
-		return WindowsLogFilePath
+		return WindowsInstallPath + "logs\\aws-otel-collector.log"
```
Comment (yimuniao) [inline pkg/logger/logger.go:78]: Looks like the "Logs" changed to "logs", it is intended?

File: pkg/userutils/userutil_linux.go (hunk 4 above)
Comment (yimuniao) [reply pkg/userutils/userutil_linux.go:79]: How to use customized user?

File: pkg/logger/logger.go (hunk 6 above)
Comment (mxiamxia) [reply pkg/logger/logger.go:78]: No, thanks for catching it.

File: pkg/userutils/userutil_linux.go (hunk 4 above)
Comment (mxiamxia) [reply pkg/userutils/userutil_linux.go:79]: added `func getCustomUser() string {}` ? but we iterate it later if read customized user from Env Var doesn't work well.

**Review by yimuniao — APPROVED** [review]

**Pull request merged**